# 3.5.10 채점 점수 구하기

In [ ]:
import json 
from glob import glob 

path = "./qa_evaluation_results"

path_list = glob(f"{path}/*")

# 두 개의 빈 리스트를 만들어 계산된 평균과 모델의 평균 점수를 저장할 준비를 함.
calculated_average_list = [] 
model_average_score = [] 

# 각 파일에 대해 반복문을 실행하며, 파일을 열어 JSON 형식의 데이터를 로드함
for path in path_list:
    with open(path, "r") as file:
        data = json.load(file)
    
    data = data["choices"][0]["message"]["parsed"]
    scores = [item['score'] for item in data.values() if isinstance(item, dict)]
    calculated_average = sum(scores) / len(scores)
    calculated_average_list.append(calculated_average)
    model_average_score.append(data["average_score"])

# 모든 파일 처리가 끝나면 두 리스트의 전체 평균을 각각 계산해 변수에 저장.
mean_calculated_average= sum(calculated_average_list) / len(calculated_average_list)
mean_model_average_score = sum(model_average_score) / len(model_average_score)
mean_calculated_average, mean_model_average_score

(5.329400000000011, 5.31807000000001)

결과를 보면 10점만점에 5.32점이라는 괜찮은 점수를 받음. 

지금까지 2번의 전체 파인튜닝 실습을 진행함. 파인튜닝은 모델의 모든 가중치를 조정하는 방식임. 하지만 이 방법에는 단점이 존재함
- **기존에 열심히 학습한 모델의 좋은 가중치가 모두 변경된다는 점**
- 이는 때로 모델의 일반적인 성능을 저하시킬 수 있음. 이를 해결하기 위해서 4장에서 LoRA 튜닝과 QLoRA 튜닝을 소개하고 실습함.